src/vm/vm.c

```diff
@@ -1361,7 +1360,12 @@ static DictuInterpretResult run(DictuVM *vm) {
             ip += offset;
             DISPATCH();
         }
-
+        CASE_CODE(COMPARE_JUMP):{
+            uint16_t offset = READ_SHORT();
+            Value a=pop(vm);
+            if (isFalsey(BOOL_VAL(valuesEqual(peek(vm,0), a)))) ip += offset;
```

Comment:
Here we can just use the value of `valuesEqual` since it returns a boolean value.

```suggestion
            if (!valuesEqual(peek(vm,0), a) ip += offset;
```